src/Features/Core/Portable/CodeCleanup/AbstractCodeCleanupService.cs

```diff
@@ -136,7 +156,92 @@ protected AbstractCodeCleanupService(ICodeFixService codeFixService)
             return document;
         }
 
+        private async Task<Document> ApplyThirdPartyCodeFixesAsync(Document document, IProgressTracker progressTracker, CodeActionOptions options, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var range = new TextSpan(0, tree.Length);
+
+            // Compute diagnostics for everything that is not an IDE analyzer
+            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range,
+                shouldIncludeDiagnostic: static diagnosticId => !(IDEDiagnosticIdToOptionMappingHelper.IsKnownIDEDiagnosticId(diagnosticId) || IsCompilerDiagnostic(diagnosticId)),
+                includeSuppressedDiagnostics: false,
+                cancellationToken: cancellationToken).ConfigureAwait(false));
+
+            // ensure more than just known diagnostics were returned
+            if (!diagnostics.Any())
+            {
+                return document;
+            }
+
+            var distinctDiagnostics = diagnostics.SelectAsArray(static d => d.Id).Distinct();
+
+            // ensure progress is reported for each third party code fix
+            progressTracker.AddItems(distinctDiagnostics.Length);
```

Comment:
adding progress after the fact isn't good as it means that progress looks like ti's about to finish, then goes *backwards*.  we always want the items up front so taht progress monotonically increases.

Reply:
updated to not do this